ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js

```diff
@@ -20,32 +35,101 @@ class BaseLine extends React.Component {
   render() {
     const props = this.props
     const { label, cursor, geometryKey, unitKey, widthKey } = props
-
     return (
-      <div className="input-location">
-        <TextField
-          label={label}
-          value={this.state.value}
-          onChange={value => {
-            this.setState({ value })
-            const fn = cursor(geometryKey)
-            try {
-              fn(JSON.parse(value))
-            } catch (e) {}
-          }}
-        />
-        <Units value={props[unitKey]} onChange={cursor(unitKey)}>
+      <React.Fragment>
+        <div className="input-location">
           <TextField
-            type="number"
-            label="Buffer width"
-            min={0.000001}
-            value={props[widthKey]}
-            onChange={cursor(widthKey)}
+            label={label}
+            value={this.state.value}
+            onChange={value => {
+              this.setState({ value })
+              const fn = cursor(geometryKey)
+              try {
+                fn(JSON.parse(value))
+              } catch (e) {}
+            }}
+            onBlur={() => this.isValidInput(this.state.value)}
+            onFocus={value => {
+              this.isValid = true
+              this.setState()
+            }}
           />
-        </Units>
-      </div>
+          <Units value={props[unitKey]} onChange={cursor(unitKey)}>
+            <TextField
+              type="number"
+              label="Buffer width"
+              min={0.000001}
+              value={props[widthKey]}
+              onChange={cursor(widthKey)}
+            />
+          </Units>
+        </div>
+        {this.isValid ? (
+          ''
+        ) : (
+          <Invalid>
+            &nbsp;
+            <span className="fa fa-exclamation-triangle" />
+            &nbsp; {this.invalidMessage}
+          </Invalid>
+        )}
+      </React.Fragment>
     )
   }
+  isValidInput(value) {
+    this.invalidMessage = ''
+    this.inValidPoint = ''
+    this.isValid = this.isValidPolygon(value)
+    this.setState({ value })
+  }
+  is2DArray(coordinates) {
```

Comment:
I feel like we could refactor this method to a utility method, because it feels like we implement it in a lot of locations